devices/AtomQrCode/QrCodeReader.cs

```diff
@@ -0,0 +1,843 @@
+        /// <summary>
+        /// Length of ACK message when decoding in host mode.
+        /// Actual data will start after this position.
+        /// </summary>
+        private static int _hostModeAckLenght = 6;
+
+        /// <summary>
+        /// Sequence common to most commands.
+        /// </summary>
+        private static byte[] _commandCommon = new byte[] { 0xC6, 0x04, 0x08, 0x00, 0xF2 };
+
+        private static byte[] _decodingCommandBase = new byte[] { 0x04, 0xE4, 0x04, 0x00, 0xFF, 0x14 };
+
+        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+        private BarcodeDataAvailableEventHandler _callbacksBarcodeDataAvailableEvent = null;
+
+        private static AutoResetEvent _newDataAvailable = new(false);
+
+        // temp buffer to hold command data
+        // created here to speed up transmission
+        private byte[] _commandBuffer;
+        private int _commandBufferLastPos = 10;
+
+        // temp buffer to hold received data
+        // created here to speed up receiving
+        private byte[] _readBuffer;
+        private int _readBufferIndex;
+
+        private bool _disposed;
+
+        private TimeSpan _lastTimeout;
+
+        // backing fields for properties
+        private readonly string _portName;
+        private readonly SerialPort _readerSerialPort;
+        private bool _waitingForData = false;
+        private TriggerMode _triggerMode = TriggerMode.None;
+
+        /// <summary>
+        /// Get and set the trigger mode for scanning.
+        /// </summary>
+        public TriggerMode TriggerMode
+        {
+            get => _triggerMode;
+            set
+            {
+                byte[] suffix = new byte[] { 0x8A, 0x08, 0xFE, 0x95 };
+
+                switch (value)
+                {
+                    case TriggerMode.Host:
+                        // this is the default on the suffix
+                        break;
+
+                    case TriggerMode.KeyHold:
+                        suffix[_param1Index] = 0x00;
+                        suffix[_param2Index] = 0x9D;
+                        break;
+
+                    case TriggerMode.KeyPress:
+                        suffix[_param1Index] = 0x02;
+                        suffix[_param2Index] = 0x9B;
+                        break;
+
+                    case TriggerMode.Continuous:
+                        suffix[_param1Index] = 0x04;
+                        suffix[_param2Index] = 0x99;
+                        break;
+
+                    case TriggerMode.Autosensing:
+                        suffix[_param1Index] = 0x09;
+                        suffix[_param2Index] = 0x94;
+                        break;
+                }
+
+                SendCommandType07(
+                    suffix,
+                    true);
+
+                // store mode
+                _triggerMode = value;
+            }
+        }
+
+        /// <summary>
+        /// Set the buzzer volume intensity.
```

Comment:
And the individual ones, it's "Sets". Now, why not having this one as a read property as well?

Reply:
No way to read the configuration from the device.

Reply:
but you have the initialization which places it in a know state. And you can only change it here. So, that should work properly?

Reply:
I'm only doing that for the properties that I need to be sure about. Are 3 at the moment.
It would be insane doing that for all existing config for the device.

Plus: I'm guessing this is stored in a eeprom of flash in the device. If I start updating these every time the device boots, I'll be wearing the chip for nothing!